internal/xds/rbac/rbac_engine.go

```diff
@@ -85,8 +95,10 @@ func (cre *ChainEngine) IsAuthorized(ctx context.Context) error {
 
 		switch {
 		case engine.action == v3rbacpb.RBAC_ALLOW && !ok:
+			cre.logRequestDetails(rpcData)
```

Comment:
Is the returned error, below, being logged by the caller?  Otherwise, you have this log message about the RPC details but without any indication that it's been rejected or whether it was an allow/deny policy (and which one, if deny) that caused it.

Reply:
Actually we do have the log here
https://github.com/grpc/grpc-go/blob/master/internal/xds/rbac/rbac_engine.go#L83
which says whether we found a match in deny or allow policy
So on finding match in deny or absence of this log suggests a denied request
But since this requires code knowledge, I added an additional log for unauthorized request. PTAL.

Reply:
SG thanks for the explanation.